src/jsutils/suggestionList.js

```diff
@@ -0,0 +1,82 @@
+/* @flow */
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/**
+ * Given an invalid input string and a list of valid options, returns a filtered
+ * list of valid options sorted based on their simularity with the input.
+ */
+export function suggestionList(
+    input: string,
+    options: Array<string>
+  ): Array<string> {
+  let i;
+  const d = {};
+  const oLength = options.length;
+  const inputThreshold = input.length / 2;
+  for (i = 0; i < oLength; i++) {
+    const distance = lexicalDistance(input, options[i]);
+    const threshold = Math.max(inputThreshold, options[i].length / 2, 1);
+    if (distance <= threshold) {
+      d[options[i]] = distance;
+    }
+  }
+  const result = Object.keys(d);
+  return result.sort((a , b) => d[a] - d[b]);
+}
+
+/**
+ * Computes the lexical distance between strings A and B.
+ *
+ * The "distance" between two strings is given by counting the minimum number
+ * of edits needed to transform string A into string B. An edit can be an
+ * insertion, deletion, or substitution of a single character, or a swap of two
+ * adjacent characters.
```

Comment:
This is copied pasta from https://github.com/graphql/codemirror-graphql/blob/master/src/utils/hintList.js#L40

I did not put too much though on which specific algorithm to use. I just found the one that will get us most of the way there. Improvements can always be made later when we understand the pitfalls of this one better. Looking at the wikipedia page for `Damerau–Levenshtein`, is "a transposition of two adjacent characters" different from "a swap of two adjacent characters" ?